xml-parser/src/main/java/net/adoptopenjdk/icedteaweb/xmlparser/NodeUtils.java

```diff
@@ -0,0 +1,202 @@
+package net.adoptopenjdk.icedteaweb.xmlparser;
+
+import net.adoptopenjdk.icedteaweb.Assert;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.adoptopenjdk.icedteaweb.i18n.Translator.R;
+
+/**
+ * ...
+ */
+public class NodeUtils {
+    /**
+     * Returns the first child node with the specified name.
+     */
+    public static Node getChildNode(final Node node, final String name) {
+        final Node[] result = getChildNodes(node, name);
+        if (result.length == 0) {
+            return null;
+        } else {
+            return result[0];
+        }
+    }
+
+    /**
+     * Returns all child nodes with the specified name.
+     */
+    public static Node[] getChildNodes(final Node node, final String name) {
+        final List<Node> result = new ArrayList<>();
+
+        Node child = node.getFirstChild();
+        while (child != null) {
+            if (child.getNodeName().getName().equals(name)) {
+                result.add(child);
+            }
+            child = child.getNextSibling();
+        }
+
+        return result.toArray(new Node[result.size()]);
```

Comment:
consider to use result.toArray(new Node[0]) here as recommended since Java6 by code analyzer